Src/HALALMock/Services/InputCapture/InputCapture.cpp

```diff
@@ -18,15 +18,12 @@ static map<uint32_t, uint32_t> channel_dict = {
 	{HAL_TIM_ACTIVE_CHANNEL_6, TIM_CHANNEL_6}
 };
 
-InputCapture::Instance::Instance(Pin& pin, TimerPeripheral* peripheral, uint32_t channel_rising, uint32_t channel_falling) :
-	pin(pin),
-	peripheral(peripheral),
-	channel_rising(channel_rising),
-	channel_falling(channel_falling)
+InputCapture::Instance::Instance(Pin& pin, void* peripheral, uint32_t channel_rising, uint32_t channel_falling) :
+	pin(pin)
 	{
-		frequency = 0;
-		duty_cycle = 0;
-
+		EmulatedPin& sim_pin = SharedMemory::get_pin(pin);
+		duty_cycle = &(sim_pin.PinData.InputCapture.duty_cycle);
+		frequency = &(sim_pin.PinData.InputCapture.frequency);
```

Comment:
set to zero here, the initialization of duty_cycle and frequency has to take place here